src/app/patient-dashboard/oncology/diagnosis-history/oncology-diagnosis-history.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, OnDestroy, OnInit, Input } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { Patient } from '../../../models/patient.model';
+import { PatientService } from '../../services/patient.service';
+import { OncologySummaryResourceService
+} from '../../../etl-api/oncology-summary-resource.service';
+
+@Component({
+  selector: 'oncology-diagnosis-history',
+  templateUrl: './oncology-diagnosis-history.component.html',
+  styles: []
+})
+export class OncologyDiagnosisHistoryComponent implements OnInit, OnDestroy {
+
+  public loadingSummary = false;
+  public subscription: Subscription;
+  public patient: Patient;
+  public patientUuid: any;
+  public errors: any = [];
+  public patientData: any;
+  public diagnosisChanges: any;
+  @Input() public programUuid;
+
+  constructor(
+    private patientService: PatientService,
+    private oncolologySummary: OncologySummaryResourceService) {
+  }
+
+  public ngOnInit() {
+    this.getPatient();
+  }
+
+  public ngOnDestroy() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  public getPatient() {
+    this.loadingSummary = true;
```

Comment:
setting of this flag to true should probably go inside loadOncologyDiagnosisHistory(), and should be the first item